_posts/2023-10-11-mcu-periphral-forwarding.md

```diff
@@ -177,17 +177,17 @@ void foo(){
 } 
 ```
 
-Using the LLVM API, we can determine the code above does not require instrumentation. By analyzing the code, LLVM can identify memory operations that correspond to local PC pointers and exclude them from the instrumentation process.
+Using the LLVM API, we can determine the code above does not require instrumentation. By analyzing the code, LLVM can identify memory operations corresponding to local PC pointers and exclude them from the instrumentation process.
 
-However, for cases where the distinction is unclear, you would need to check every memory address against a set of peripheral address intervals. For example, in the case of the STM32L0 chip, the peripheral registers are located within specific address intervals, such as 0x40000000 - 0x40008000, 0x40010000 - 0x40018000, 0x4002 0000 - 0x4002 63FF and 0x5000 0000 - 0x5000 1FFF. By comparing the accessed address with these intervals, you can determine if it corresponds to a peripheral address. 
+However, for cases where the distinction is unclear, you must check every memory address against a set of peripheral address intervals. For example, in the case of the STM32L0 chip, the peripheral registers are located within specific address intervals, such as 0x40000000 - 0x40008000, 0x40010000 - 0x40018000, 0x4002 0000 - 0x4002 63FF and 0x5000 0000 - 0x5000 1FFF. By comparing the accessed address with these intervals, you can determine if it corresponds to a peripheral address. 
 
 ![STM32L0 memory map](/img/chip-peripheral-forwarding/memory_map.png)
 
 If the checking address falls within one of the defined peripheral address intervals, it can be considered a peripheral address. Conversely, if the checking address falls outside these intervals, it is likely a local host address.
 
 ### AddressInterceptor LLVM Transform pass
 
-I have developed an [LLVM Transform pass](https://llvm.org/docs/WritingAnLLVMPass.html#introduction-what-is-a-pass)(plugin) called [AddressInterceptor](https://github.com/remotemcu/adin-llvm-pass) (ADIN) to address the challenge of distinguishing between local and global variables and peripheral addresses during instrumentation. ADIN performs checks on local and global variables to determine whether an operation involves them. If it is determined that an operation does not involve local or global variables, ADIN proceeds to instrument the subsequent routines:
+I have developed an [LLVM Transform pass](https://llvm.org/docs/WritingAnLLVMPass.html#introduction-what-is-a-pass)(plugin) called [AddressInterceptor](https://github.com/remotemcu/adin-llvm-pass) (ADIN) to address the challenge of distinguishing between local and global variables and peripheral addresses during instrumentation. ADIN checks local and global variables to determine whether an operation involves them. If an operation does not, ADIN proceeds to instrument the subsequent routines:
```

Comment:
I'm concerned that the original meaning might have shifted. Perhaps we should consider replacing the paragraph with this version:
"
I have developed an [LLVM Transform pass](https://llvm.org/docs/WritingAnLLVMPass.html#introduction-what-is-a-pass)(plugin) called [AddressInterceptor](https://github.com/remotemcu/adin-llvm-pass) (ADIN) to address the challenge of distinguishing between PC application addresses and peripheral addresses during instrumentation. ADIN checks all memory operations, assessing whether each operation involves global or local variables of the application. If an operation does not align with these types, ADIN proceeds to instrument the subsequent routines:
"

Reply:
Done! Adjusted [here](https://github.com/memfault/interrupt/pull/385/commits/0260671fff2a69d12cba575e6f86408163e1b42b). Used "involves" instead of "aligns" but otherwise is word-for-word your changes.